src/uu/expr/src/syntax_tree.rs

```diff
@@ -155,8 +155,24 @@ impl AstNode {
         }
     }
     pub fn operand_values(&self) -> Result<Vec<String>, String> {
-        if let Self::Node { operands, .. } = self {
+        if let Self::Node {
+            operands, op_type, ..
+        } = self
+        {
             let mut out = Vec::with_capacity(operands.len());
+            let mut operands = operands.iter();
+
+            if op_type == "|" {
+                if let Some(value) = operands.next() {
```

Comment:
could you please add some comments to describe this ? 

and explain what it is doing :)

thanks

Reply:
I have add a comment to describe it :)